### PR TITLE
Constant TCP velocity time parameterization

### PR DESCRIPTION
**Pull request by marip8**
This PR implements a time parameterization algorithm for maintaining a constant TCP speed. The approach works by:
- Calculating FK at each waypoint in the trajectory to get the pose of the TCP
- Creating a line path between adjacent TCP poses
- Parameterize the line with a trapezoidal velocity profile
- Compute the joint velocity/acceleration at each waypoint given the TCP Cartesian velocity/acceleration
- Validate that the joint velocity/acceleration is within the limits of the robot

# Before
https://github.com/ros-industrial-consortium/scan_n_plan_workshop/assets/18411310/9e829756-0953-4379-be9b-b353c9112faf

# After
https://github.com/ros-industrial-consortium/scan_n_plan_workshop/assets/18411310/77bbd6f7-bb43-4043-910b-714a9963e6f9

### CODE REVIEW

**Comment by DavidSpielman** [comment]
https://github.com/ros-industrial-consortium/scan_n_plan_workshop/assets/76972296/2236552d-ffac-4c64-91d3-2090b0e29e6b

While testing various tool paths, I came across this one. I am not sure what is causing the sporadic back and forth motion of the robot.

**Comment by marip8** [comment]
> unusual_motion_plan.mp4 
> While testing various tool paths, I came across this one. I am not sure what is causing the sporadic back and forth motion of the robot. The rest of the motions were smooth as expected.

I think this is actually a tool path issue, not a time parameterization issue. It looks like there is a small tool path that was created in the mesh that caused this

File: snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    timer.start();
+    saveInputs(*info, input);
+
+    // --------------------
+    // Check that inputs are valid
+    // --------------------
+    tesseract_planning::Instruction* input_results = input.getResults();
+    if (!isCompositeInstruction(*input_results))
+    {
+      CONSOLE_BRIDGE_logError("Input results to iterative spline parameterization must be a composite instruction");
```
Comment (marrts) [inline snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp:55]: Should reference Cartesian not iterative spline

File: snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    tesseract_planning::TrajectoryContainer::Ptr trajectory =
+        std::make_shared<tesseract_planning::InstructionsTrajectory>(ci);
+
+    CartesianTimeParameterization solver(input.env, manip_info.manipulator, manip_info.tcp_frame,
+                                         profile->max_translational_velocity, profile->max_rotational_velocity,
+                                         profile->max_translational_acceleration, profile->max_rotational_acceleration);
+
+    if (!solver.compute(*trajectory, profile->max_velocity_scaling_factor, profile->max_acceleration_scaling_factor))
+    {
+      CONSOLE_BRIDGE_logInform("Failed to perform iterative spline time parameterization for process input: %s!",
```
Comment (marrts) [inline snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp:99]: Same about reference to ISP

File: snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                               input_results->getDescription().c_str());
+      saveOutputs(*info, input);
+      info->elapsed_time = timer.elapsedSeconds();
+      return 0;
+    }
+
+    CONSOLE_BRIDGE_logDebug("Iterative spline time parameterization succeeded");
```
Comment (marrts) [inline snp_motion_planning/src/cartesian_time_parameterization_task_generator.hpp:106]: Same about ISP

File: snp_motion_planning/src/cartesian_time_parameterization.hpp [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    return q_dot_dot;
+  }
+
+  tesseract_kinematics::JointGroup::UPtr motion_group;
+  KDL::Chain kdl_chain_;
+  const std::string tcp;
+  const double max_translational_vel;
+  const double max_translational_acc;
```
Comment (marrts) [inline snp_motion_planning/src/cartesian_time_parameterization.hpp:275]: I feel like these should have an underscore suffix as they are class members. Same with motion_group. It's confusing in the constructor.

**Pull request merged**